src/amp-react-selector.js

```diff
@@ -0,0 +1,142 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import ReactCompatibleBaseElement from './react-compat-base-element.js';
+import {
+  AmpContext,
+} from './amp-context.js';
+import {Slot} from './slot.js';
+
+const {
+  useContext,
+  useEffect,
+  useLayoutEffect,
+  useRef,
+  useState,
+} = preactHooks;
+
+
+export const AmpSelectorContext = preact.createContext();
+
+
+export function AmpSelector(props) {
+  const {multiple, value, defaultValue} = props;
+  const [selected, setSelectedState] = useState(
+    value ? [].concat(value) :
+    defaultValue ? [].concat(defaultValue) :
+    []);
+  const setSelectedRef = useRef(value => {
+    if (props.onChange) {
+      props.onChange({target: {value: multiple ? value : value[0]}});
+    }
+    setSelectedState(value);
+  });
+  return preact.createElement(
+    props.tagName || 'div',
+    {
+      ...props,
+    },
+    preact.createElement(
+      AmpSelectorContext.Provider,
+      {
+        ...props,
+        value: {
+          multiple,
+          // TBD: controlled values require override of properties.
+          selected: value ? [].concat(value) : selected,
+          setSelected: setSelectedRef.current,
+        },
+      }
+    )
+  );
+}
+
+AmpSelector.Option = function(props) {
+  const {option} = props;
+  const selectorContext = useContext(AmpSelectorContext);
+  const {selected, setSelected, multiple} = selectorContext;
+  const optionProps = {
+    ...props,
+    option,
+    selected: selected.includes(option),
+    onClick: () => selectOption(option, multiple, selected, setSelected),
+  };
+  if (props.render) {
+    return props.render(optionProps);
+  }
+  return preact.createElement(
+    props.type || props.tagName || 'div',
+    optionProps,
+    props.children);
+}
+
+function selectOption(option, multiple, selected, setSelected) {
+  if (multiple) {
+    setSelected(
+      selected.includes(option) ?
+      selected.filter(v => v != option) :
+      selected.concat(option));
+  } else {
+    if (!selected.includes(option)) {
+      setSelected([option]);
+    }
+  }
+}
+
+
+const AmpReactSelector = ReactCompatibleBaseElement(AmpSelector, {
+  className: 'i-amphtml-fill-content i-amphtml-replaced-content',
+  attrs: {
+  },
+  template: true,
+  // TODO: simply override ReactBaseElement instead.
+  init(reactBaseElement) {
+    const {element} = reactBaseElement;
+    const mu = new MutationObserver(records => {
+      rebuild();
+    });
+    const rebuild = () => {
+      const children = [];
+      const optionChildren = element.querySelectorAll('[option]');
+      const value = [];
+      optionChildren.forEach(child => {
+        // TODO: check that an option is not within another option.
+        const option = child.getAttribute('option');
+        const props = {
+          option,
+          type: Slot,
```

Comment:
Should we use tagName here, too?

Reply:
`Slot` component defines its own, so there's no need to repeat it here. This is a predefined amp-element/component binding, so we don't need too many variances between them. I.e. in this case AMP would always bind a slot to the selector option.